test/browser.js

```diff
@@ -236,3 +243,34 @@ test('headers are preserved when input is a Request and there are searchParams i
 
 	await server.close();
 });
+
+test('retry with body', withPage, async (t, page) => {
+	let requestCount = 0;
+
+	const server = await createTestServer();
+	server.get('/', (request, response) => {
+		response.end('zebra');
+	});
+	server.put('/test', async (request, response) => {
+		requestCount++;
+		await pBody(request);
+		response.sendStatus(502);
+	});
+
+	await page.goto(server.url);
+	await page.addScriptTag({path: './umd.js'});
+
+	const error = await page.evaluate(url => {
+		window.ky = window.ky.default;
+		const request = window.ky(url + '/test', {
+			body: 'foo',
+			method: 'PUT',
+			retry: 2
+		}).text();
+		return request.catch(error_ => error_.toString());
+	}, server.url);
+	t.is(error, 'HTTPError: Bad Gateway');
+	t.is(requestCount, 2);
```

Comment:
For reasons that I do not understand, the `requestCount` assertion was failing and I was only able to solve it by changing the status code from `408` to something else. When the status is `408`, the server receives roughly double the number of requests I am expecting (i.e. roughly 2x whatever `retry` is). When I set the `retry` option to `1`, then `requestCount` was `2` and the assertion passed. When I set `retry` to `8`, then `requestCount` was `14` and the assertion failed (note: not quite double).

So, of course it seems like a bug in Ky. And sure enough, when I debugged it with `await puppeteer.launch({ devtools: true });` in [test/helpers/with-page.js](https://github.com/sindresorhus/ky/blob/78814e74efaaab65728103a13541820441b05d93/test/helpers/with-page.js#L4), I was able to see these requests in the Network tab and the timings looked like they were respecting our [backoff algorithm](https://github.com/sindresorhus/ky/blob/78814e74efaaab65728103a13541820441b05d93/index.js#L337-L338), so these requests are likely being triggered within Ky.

The thing is, we have no special handling of `408`, nor `502` in Ky, so I don't see how this could be fixed by changing from `408` to `502`. I also wasn't able to easily reproduce this behavior manually in Chrome using endpoints from https://httpstat.us/. To me, it's a mystery.

I chose to just change the status code in commit https://github.com/sindresorhus/ky/pull/231/commits/2808b0fb953688cf7fdc04972822428581188582.